src/coreclr/vm/siginfo.cpp

```diff
@@ -386,6 +386,20 @@ void SigPointer::ConvertToInternalSignature(Module* pSigModule, SigTypeContext *
         cArgs--;
     }
 }
+
+void SigPointer::CopySignature(Module* pSigModule, SigBuilder* pSigBuilder, BYTE additionalCallConv)
+{
+    CONTRACTL
+    {
+        INSTANCE_CHECK;
+        STANDARD_VM_CHECK;
+    }
+    CONTRACTL_END
+
+    // The copy is done without advancing m_ptr.
+    pSigBuilder->AppendByte(*m_ptr | additionalCallConv);
+    pSigBuilder->AppendBlob((const PVOID)(m_ptr + 1), m_dwLen - 1);
```

Comment:
This is going to copy everything in the SigPointer, not just the signature. It does not seem like a good idea. I am not sure about where it may cause things to go wrong.

I think you should only copy what's required. https://github.com/dotnet/runtime/pull/114495/#discussion_r2040391681 was my suggestion for how to do it.

Reply:
Yep sorry regressed here. I was trying to avoid a second skip\parse of the signature; the current position of the next unprocessed is there in MetaSig but it doesn't expose it (it only exposes the last start of argument) probably to help detect invalid sigs or usage. Also updated the calling code a bit to make it more obvious when we skip the element type.